### DISCUSSION TITLE
Sky models in Mitsuba3

### DISCUSSION THREAD

**Discussion by imanolooo** [mitsuba-renderer/mitsuba3#1193] [Q&A] [answered]
Hi all,

I have seen that in Mitsuba1 there were available sky models. However, I don't see they are available in Mitsuba3. Am I right?

I would like to make some renders for indoor scenes, combining several spectral emitters and natural lighting from the windows. I'm not really sure on how to deal with that in Mitsuba3. Is it possible to do spectral rendering for natural lighting? I have seen I could use constant environment emitters for the sky and distant directional emitter for the sun. However, in this case, I would lose the changes on the spectrum through the sky hemisphere...

Another option I have seen is to use environment maps. Does Mitsuba support to load OpenEXR with as many channels as spectral bands? Or it is only thought to RGB HDR images?

Thanks a lot! Sorry, if I am missing some point and it is easier.

Best,

Imanol

**Comment by njroussel** [comment, marked as answer]
Hi @imanolooo 

Indeed, they have not been ported to MItsuba 3.

The current `envmap` only support `spectral` rendering by using a rgb->spectral upsampling scheme. You'd need to modify the plugin if you want to provide with your own raw spectral measurements.

**Reply by imanolooo** [reply]
Ok! Thanks a lot!

If I succeed doing I will let you know!

Best,

Imanol
